workflow/workflow.go

```diff
@@ -607,6 +623,34 @@ func SetCurrentDetails(ctx Context, details string) {
 	internal.SetCurrentDetails(ctx, details)
 }
 
+// SetVersioningBehavior sets the strategy to upgrade this workflow when the default Build ID
+// has changed, and Worker Versioning-3 has been enabled.
+//
+// SetVersioningBehavior should be called during the first workflow task, and the context `ctx` should
```

Comment:
What if it is called, but not in the first workflow task?

Reply:
If there was no worker level default, the server will fail the first workflow task, so there is no progress until a worker sets a default.
If tthere was a worker level default, it will change it ok.
In the comment I'm encouraging to set it in the first task to avoid the first case...